web-registry/src/hooks/useQueryClient.ts

```diff
@@ -0,0 +1,104 @@
+import { useState, useEffect } from 'react';
+
+import { QueryClientImpl as BankQueryClient } from '@regen-network/api/lib/generated/cosmos/bank/v1beta1/query';
+import { QueryClientImpl as EcocreditQueryClient } from '@regen-network/api/lib/generated/regen/ecocredit/v1alpha1/query';
+import { QueryClientImpl as BasketQueryClient } from '@regen-network/api/lib/generated/regen/ecocredit/basket/v1/query';
+
+import { useLedger } from '../ledger';
+
+// TODO: map clients and ledger queries here, hidding this details from business hooks
+// TODO: Map the clients and queries here, so that the implementation details
+// of the queries are transparent to business hooks. Instead, the hook would be called
+// like this:
+// ie. useQueryClient('basket', { basketDenom })
+// ie. useQueryClient('denomMetadata', { denom: basketDenom })
+
+type QueryClient = BankQueryClient | EcocreditQueryClient | BasketQueryClient;
+
+type LedgerModules = 'bank' | 'ecocredit' | 'basket';
+
+// TODO: generic type
+type Ledgerparams = any;
+
+// type LedgerQuery<P> = ({
+//   client,
+//   params,
+// }: {
+//   client: QueryClient;
+//   params: Ledgerparams<P>;
+// }) => Promise<any>;
+
+// type LedgerQuery = ({
+//   client,
+//   params,
+// }: {
+//   client: QueryClient;
+//   params: Ledgerparams;
+// }) => Promise<any>;
+
+// TODO: generic types
+type LedgerQuery = ({ client, params }: any) => Promise<any>;
+
+type InputQueryLedger = {
+  type: LedgerModules;
```

Comment:
`module` is more expressive here, since it is very relevant in the Cosmos world.
```suggestion
  module: LedgerModule;
```

Reply:
agreed

Reply:
yes, totally agree. I have also changed the name of the callback parameter to query, following the same logic